core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -148,6 +148,9 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
       charset: Charset = DefaultCharset): EntityEncoder[F, Array[Char]] =
     stringEncoder[F].contramap(new String(_))
 
+  implicit def segmentEncoder[F[_]: Applicative]: EntityEncoder[F, Segment[Byte, Unit]] =
+    chunkEncoder[F].contramap[Segment[Byte, Unit]](_.force.toChunk) //TODO: Seems fishy
```

Comment:
Maybe this is what sucks because of the definition of `simple`.

I don't yet have a strong opinion on whether we need an encoder for both `Segment` and `Chunk`.

Reply:
If we define both a decoder and an encoder, we get an extra test for free in the entitycodec spec is all, so I figured this would be okay.